### PR TITLE
Taint based eviction promoted to GA in 1.18

### PR DESCRIPTION
**Pull request by ingvagabund**
Taint based eviction as described in https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/#taint-based-evictions is about to be GA'ed in 1.18. Updating the docs appropriately.

Fixes: https://github.com/kubernetes/kubernetes/issues/87429

Will cherry-pick to 1.18 until this one is reviewed or before Friday.

### CODE REVIEW

**Comment by damemi** [comment]
/hold
until the other PRs go in to functionally make this GA

**Comment by sftim** [comment]
Hi @ingvagabund 

This repository has a live _master_ branch (and does not usually cherry-pick). I think you should edit this PR to target the branch _dev-1.18_.

/sig scheduling

**Comment by ingvagabund** [comment]
@sftim thanks!!!

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -456,8 +454,6 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `SupportPodPidsLimit`: Enable the support to limiting PIDs in Pods.
 - `Sysctls`: Enable support for namespaced kernel parameters (sysctls) that can be set for each pod.
   See [sysctls](/docs/tasks/administer-cluster/sysctl-cluster/) for more details.
-- `TaintBasedEvictions`: Enable evicting pods from nodes based on taints on nodes and tolerations on Pods.
-  See [taints and tolerations](/docs/concepts/configuration/taint-and-toleration/) for more details.
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Leave these lines in

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -138,8 +138,6 @@ different Kubernetes components.
 | `SupportPodPidsLimit` | `false` | Alpha | 1.10 | 1.13 |
 | `SupportPodPidsLimit` | `true` | Beta | 1.14 | |
 | `Sysctls` | `true` | Beta | 1.11 | |
-| `TaintBasedEvictions` | `false` | Alpha | 1.6 | 1.12 |
-| `TaintBasedEvictions` | `true` | Beta | 1.13 | |
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]: @ingvagabund, I think you want to **move** these lines into the section _Feature gates for graduated or deprecated features_.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Leave these lines in

File: content/en/docs/concepts/configuration/taint-and-toleration.md [hunk 3]
```diff
@@ -229,7 +229,7 @@ certain condition is true. The following taints are built in:
     as unusable. After a controller from the cloud-controller-manager initializes
     this node, the kubelet removes this taint.
 
-In version 1.13, the `TaintBasedEvictions` feature is promoted to beta and enabled by default, hence the taints are automatically
+In version 1.18, the `TaintBasedEvictions` feature is promoted to GA and enabled by default, hence the taints are automatically
 added by the NodeController (or kubelet) and the normal logic for evicting pods from nodes
 based on the Ready NodeCondition is disabled.
```
Comment (sftim) [inline content/en/docs/concepts/configuration/taint-and-toleration.md]: Consider rewording this to suit a graduated feature: imagine that Kubernetes v1.23 is out and this text is in place; does it look out of place?

Comment (damemi) [reply content/en/docs/concepts/configuration/taint-and-toleration.md]: @sftim do you mean remove the specific reference to 1.18? I think it makes sense to mention what version it was GA'd in for future reference, so maybe something like `As of version 1.18, ...`?

Comment (sftim) [reply content/en/docs/concepts/configuration/taint-and-toleration.md]: How about adding a feature-state shortcode on line 204 or thereabouts?

`{{< feature-state for_k8s_version="1.18" state="stable" >}}`

Comment (sftim) [reply content/en/docs/concepts/configuration/taint-and-toleration.md]: Try it! Add it and you'll see the result in the local preview.
or there's an example at https://kubernetes.io/docs/contribute/style/hugo-shortcodes/#feature-state

Comment (ingvagabund) [reply content/en/docs/concepts/configuration/taint-and-toleration.md]: Done

**Comment by VineethReddy02** [comment]
👋 Please rebase this PR on `dev-1.18` in order to avoid merge conflicts.

Feel free to `/hold cancel` after you've rebased.

/hold

File: content/en/docs/concepts/configuration/taint-and-toleration.md [hunk 4]
```diff
@@ -277,15 +278,14 @@ admission controller](https://git.k8s.io/kubernetes/plugin/pkg/admission/default
...
-which matches the behavior when this feature is disabled.
+This ensures that DaemonSet pods are never evicted due to these problems.
 
 ## Taint Nodes by Condition
 
 The node lifecycle controller automatically creates taints corresponding to
 Node conditions.
 Similarly the scheduler does not check Node conditions; instead the scheduler checks taints. This assures that Node conditions don't affect what's scheduled onto the Node. The user can choose to ignore some of the Node's problems (represented as Node conditions) by adding appropriate Pod tolerations.
-Note that `TaintNodesByCondition` only taints nodes with `NoSchedule` effect. `NoExecute` effect is controlled by `TaintBasedEviction` which is a beta feature and enabled by default since version 1.13.
+Note that `TaintNodesByCondition` only taints nodes with `NoSchedule` effect. `NoExecute` effect is controlled by `TaintBasedEviction`.
```
Comment (sftim) [inline content/en/docs/concepts/configuration/taint-and-toleration.md]: You should reword this to avoid mentioning the names of the two now-graduated feature gates.

File: content/en/docs/concepts/configuration/taint-and-toleration.md [hunk 5]
```diff
@@ -229,9 +231,8 @@ certain condition is true. The following taints are built in:
     as unusable. After a controller from the cloud-controller-manager initializes
     this node, the kubelet removes this taint.
 
-In version 1.13, the `TaintBasedEvictions` feature is promoted to beta and enabled by default, hence the taints are automatically
-added by the NodeController (or kubelet) and the normal logic for evicting pods from nodes
-based on the Ready NodeCondition is disabled.
+The taints are automatically added by the NodeController (or kubelet) and the normal
+logic for evicting pods from nodes based on the Ready NodeCondition is disabled.
```
Comment (sftim) [inline content/en/docs/concepts/configuration/taint-and-toleration.md]: Is this also accurate
```suggestion
When the node controller or the kubelet observe problems on a node, they add taints to the
relevant Node object. If the fault condition returns to normal the kubelet or node
controller can remove the relevant taint(s). The kubelet reports problems that affect
its ability to run Pods whereas the node controller monitors whether the kubelet
itself is healthy.
```
?

I wouldn't mention “the normal logic for evicting pods from nodes based on the Ready NodeCondition is disabled” if there's no way to make use of that logic in v1.18. If there _is_ a way to make use of that formerly-normal logic, tell readers how to learn about that.

Comment (ingvagabund) [reply content/en/docs/concepts/configuration/taint-and-toleration.md]: We need to be careful here to properly distinguish when a taint has `NoSchedule` effect and when it has `NoExecute`. I have integrated the first two sentences. The last one expresses a general statement saying "a node is tainted based on a condition" which can have both effects interchangeable.

**Comment by ingvagabund** [comment]
> Please rebase this PR on dev-1.18 in order to avoid merge conflicts.

Done

> Feel free to /hold cancel after you've rebased.

/hold cancel

File: content/en/docs/concepts/configuration/taint-and-toleration.md (hunk 4 above)
Comment (ingvagabund) [reply content/en/docs/concepts/configuration/taint-and-toleration.md]: I removed those two lines completely and move mention of each effect to corresponding section.

**Comment by ingvagabund** [comment]
@sftim updated, PTAL

**Comment by damemi** [comment]
@sftim thanks, can you /approve this?

**Comment by sftim** [comment]
@damemi I'm not on the docs release team
@VineethReddy02 would you have time to take a look?

**Comment by sftim** [comment]
/approve
following https://github.com/kubernetes/website/pull/19302#issuecomment-598376036
